saltproc/abc.py

```diff
@@ -0,0 +1,163 @@
+from abc import ABC, abstractmethod
+
+class Depcode(ABC):
+    """Abstract class for interfacing with monte-carlo particle transport
+    codes. Contains information about input, output, geometry, and template
+    files for running depletion simulations. Also contains neutron
+    population, active, and inactive cycles. Contains methods to read template
+    and output files, and write new input files for the depletion code.
+
+    Attributes
+    -----------
+    param : dict of str to type
+        Holds depletion step parameter information. Parameter names are keys
+        and parameter values are values.
+    sim_info : dict of str to type
+        Holds simulation settings information. Setting names are keys
+        and setting values are values.
+    iter_inputfile : str
+        Path to depletion code input file for depletion code rerunning.
+    iter_matfile : str
+        Path to iterative, rewritable material file for depletion code
+        rerunning. This file is modified during  the simulation.
+
+    """
+
+    def __init__(self,
+                 codename,
+                 exec_path,
+                 template_input_file_path,
+                 geo_files=None,
+                 npop=50,
+                 active_cycles=20,
+                 inactive_cycles=20):
+        """Initializes the Depcode object.
+
+           Parameters
+           ----------
+           codename : str
+               Name of depletion code.
+           exec_path : str
+               Path to depletion code executable.
+           template_input_file_path : str or dict of str to str
+               Path(s) to depletion code input file(s), or a dictionary where
+               the input type (e.g. material, geometry, settings, etc.) as a
+               string are keys, and the path to the input file are values.
+               Type depends on depletion code in use.
+           geo_files : str or list, optional
+               Path to file that contains the reactor geometry.
+               List of `str` if reactivity control by
+               switching geometry is `On` or just `str` otherwise.
+           npop : int, optional
+               Size of neutron population per cycle for Monte Carlo.
+           active_cycles : int, optional
+               Number of active cycles.
+           inactive_cycles : int, optional
+               Number of inactive cycles.
+
+        """
+        self.codename = codename
+        self.exec_path = exec_path
+        self.template_input_file_path = template_input_file_path
+        self.geo_files = geo_files
+        self.npop = npop
+        self.active_cycles = active_cycles
+        self.inactive_cycles = inactive_cycles
+        self.param = {}
+        self.sim_info = {}
+        self.iter_inputfile = './iter_input'
+        self.iter_matfile = './iter_mat'
```

Comment:
Are these files created dynamically?

Reply:
Yes, and their names will be `inter_input` and `iter_mat` respectively. They are hardcoded because they don't really need to be seen by the user.